osidb/models.py

```diff
@@ -3293,11 +3293,11 @@ def _validate_article_links_count_via_flawreferences(self):
         article_links = self.flaw.references.filter(
             type=FlawReference.FlawReferenceType.ARTICLE
         )
-
-        if article_links.count() > 1:
+        article_count = 1 if self._state.adding else 0
+        article_count += article_links.count()
+        if article_count > 1:
             raise ValidationError(
-                f"A flaw has {article_links.count()} article links, "
-                f"but only 1 is allowed."
+                f"A flaw has {article_count} article links, " f"but only 1 is allowed."
```

Comment:
f"A flaw has {article_count} article links, but only 1 is allowed."

Reply:
nice catch, thanks!